File: Java/1370_Increasing_Decreasing_String.java
class Solution {
    int[] t = new int[26];
    StringBuilder result = new StringBuilder();
    
    public String sortString(String s) {
        for(int i = 0; i < s.length(); i++){
            t[s.charAt(i)-'a']++;
        }
        
        while(result.length() < s.length()){
            helper(true);
            helper(false);
        }
        return result.toString();
    }
    private void helper(boolean flag){
        int start = flag ? 0: 25;
        int d = flag ? 1: -1;
        for(int i = start; i < 26 && i >= 0; i+=d){
            if(t[i] != 0){
                result.append((char)('a'+i));
                t[i]--;
            }
        }
    }
}

File: CS/1370_Increasing_Decreasing_String.cs
public class Solution {
    public string SortString(string s) {
        int[] t = new int[26];
        foreach(char c in s){
            t[c-'a']++;
        }
        StringBuilder result = new StringBuilder("");
        while(result.Length != s.Length){
            for(int i = 0; i < 26; i++){
                if(t[i] > 0){
                    result.Append((char)('a' + i));
                    t[i]--;
                }
            }
            for(int i = 25; i >= 0; i--){
                if(t[i] > 0){
                    result.Append((char)('a' + i));
                    t[i]--;
                }
            }
        }
        return result.ToString();
    }
}
